File: go/internal/replay/parserfixture/testdata/trees/scala/OrderService.scala
package com.example.orders

class OrderService(repository: OrderRepository) {

  def place(customerId: String, skus: List[String]): Order = {
    val order = repository.create(customerId)
    skus.foreach(sku => order.addLine(sku))
    repository.save(order)
  }

  def find(orderId: String): Option[Order] =
    repository.byId(orderId)
}

File: go/internal/replay/parserfixture/testdata/trees/rust/order_service.rs
use crate::repository::{Order, OrderRepository};

pub struct OrderService {
    repository: OrderRepository,
}

impl OrderService {
    pub fn new(repository: OrderRepository) -> Self {
        OrderService { repository }
    }

    pub fn place(&self, customer_id: &str, skus: &[String]) -> Order {
        let mut order = self.repository.create(customer_id);
        for sku in skus {
            order.add_line(sku);
        }
        self.repository.save(order)
    }

    pub fn find(&self, order_id: &str) -> Option<Order> {
        self.repository.by_id(order_id)
    }
}
